plinth/modules/sso/__init__.py

```diff
@@ -0,0 +1,85 @@
+#
+# This file is part of Plinth.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Plinth module to configure Single Sign On services.
+"""
+import base64
+import datetime
+
+
+from plinth import action_utils
+from django.utils.translation import ugettext_lazy as _
+from plinth import actions
+from OpenSSL import crypto
+from django.utils import timezone
+
+version = 1
+
+is_essential = True
+
+depends = ['security']
+
+title = _('Single Sign On')
+
+managed_packages = ['libapache2-mod-auth-pubtkt', 'openssl', 'python3-openssl']
+
+
+def setup(helper, old_version=None):
+    """Install the required packages"""
+    helper.install(managed_packages)
+    helper.call('create-key-pair', actions.superuser_run,
+                'auth-pubtkt', ['create-key-pair'])
+    action_utils.service_restart('apache2')
```

Comment:
If this is needed, it should be done inside an action script. (Plinth might not have permission to restart a service itself.)